test/relations.test.js

```diff
@@ -5327,25 +5494,44 @@ describe('relations', function() {
            });
     });
 
-    it('should find items on scope with promises', function(done) {
+    bdd.itIf(connectorCapabilities.adhocSort !== false,
+    'should find items on scope with promises', function(done) {
       Category.findOne()
       .then(function(cat) {
-        cat.jobIds.should.eql([job2.id]);
+        cat.jobIds[0].should.eql(job2.id);
         return cat.jobs.getAsync();
       })
       .then(function(jobs) {
         var p = jobs[0];
-        p.id.should.eql(job2.id);
+        should.equal(p.id, job2.id);
         p.name.should.equal('Job 2');
         done();
       })
       .catch(done);
     });
 
+    bdd.itIf(connectorCapabilities.adhocSort === false,
+    'should find items on scope with promises', function(done) {
+      var theExpectedIds = [job1.id, job2.id, job3.id];
+      var theExpectedNames = ['Job 1', 'Job 2', 'Job 3'];
+      Category.findOne()
+      .then(function(cat) {
+        cat.jobIds[0].should.be.oneOf(theExpectedIds);
+        return cat.jobs.getAsync();
+      })
+      .then(function(jobs) {
+        var p = jobs[0];
+        p.id.should.be.oneOf(theExpectedIds);
+        p.name.should.be.oneOf(theExpectedNames);
+        done();
+      })
+      .catch(done);
+    });
+
     it('should find items on scope with promises - findById', function(done) {
       Category.findOne()
       .then(function(cat) {
-        cat.jobIds.should.eql([job2.id]);
+        cat.jobIds[0].should.eql(job2.id);
```

Comment:
```diff
 -        cat.jobIds.should.eql([job2.id]);
 +        cat.jobIds[0].should.eql(job2.id);
```

This is changing the test in a subtle way that may backfire on us in the future. Previously, the test verified that exactly one given id is returned. Now the test will verify that at least one id was returned and the first returned id is the expected one.

There are many more places in this test file where a similar change is made, please search for them yourself.